### PR TITLE
[ACCSEC-18905] Distribute sample app

### PR DESCRIPTION
**Pull request by SergioFierro**
## Jira Ticket
- ACCSEC-18905

## Description
- Distribute sample app via firebase app distribution in all 3 environments

### CODE REVIEW

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.43%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.38%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.38%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.38%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.38%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.38%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.38%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.38%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift [hunk 1]
```diff
@@ -166,7 +166,7 @@ public class TwilioVerifyBuilder {
     networkProvider = NetworkAdapter()
     jwtGenerator = JwtGenerator(withJwtSigner: JwtSigner())
     authentication = AuthenticationProvider(withJwtGenerator: jwtGenerator)
-    self.baseURL = Constants.baseURL
+    self._baseURL = baseURL
```
Comment (yafuquen) [inline TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift:169]: Is this change needed?

File: TwilioVerifyDemo/TwilioVerifyDemo.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -590,10 +590,10 @@
 			buildSettings = {
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
 				CODE_SIGN_ENTITLEMENTS = TwilioVerifyDemo/TwilioVerifyDemo.entitlements;
-				CODE_SIGN_IDENTITY = "iPhone Developer";
+				CODE_SIGN_IDENTITY = "iPhone Distribution";
 				CODE_SIGN_STYLE = Manual;
 				CURRENT_PROJECT_VERSION = 1;
-				DEVELOPMENT_TEAM = "";
+				DEVELOPMENT_TEAM = 9EVH78F4V4;
```
Comment (yafuquen) [inline TwilioVerifyDemo/TwilioVerifyDemo.xcodeproj/project.pbxproj:596]: The development team can change for someone running the sample app, can we keep it empty?

File: TwilioVerifyDemo/TwilioVerifyDemo.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -590,10 +590,10 @@
 			buildSettings = {
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
 				CODE_SIGN_ENTITLEMENTS = TwilioVerifyDemo/TwilioVerifyDemo.entitlements;
-				CODE_SIGN_IDENTITY = "iPhone Developer";
+				CODE_SIGN_IDENTITY = "iPhone Distribution";
```
Comment (yafuquen) [inline TwilioVerifyDemo/TwilioVerifyDemo.xcodeproj/project.pbxproj:593]: The developer could start running in debugging mode, can we keep it as developer?

File: TwilioVerifyDemo/TwilioVerifyDemo.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -602,7 +602,7 @@
 				MARKETING_VERSION = 1.0;
 				PRODUCT_BUNDLE_IDENTIFIER = com.twilio.TwilioVerifyDemo;
 				PRODUCT_NAME = "$(TARGET_NAME)";
-				PROVISIONING_PROFILE_SPECIFIER = "";
+				PROVISIONING_PROFILE_SPECIFIER = "match AdHoc com.twilio.TwilioVerifyDemo";
```
Comment (yafuquen) [inline TwilioVerifyDemo/TwilioVerifyDemo.xcodeproj/project.pbxproj:605]: Can we set the provisioning profile when building the apps?

File: fastlane/Fastfile [hunk 5]
```diff
@@ -86,7 +90,6 @@ platform :ios do
 
   desc "Verify next release"
   lane :verify do |options|
-    ensure_git_status_clean
```
Comment (yafuquen) [inline fastlane/Fastfile:89]: Why was this removed?

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.38%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.38%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.38%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.38%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.38%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.38%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.43%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.38%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.38%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.38%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.38%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.38%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.38%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.38%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift (hunk 1 above)
Comment (SergioFierro) [reply TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift]: Yes, because baseURL is a global variable in `TwilioVerifyConfig.swift`

File: TwilioVerifyDemo/TwilioVerifyDemo.xcodeproj/project.pbxproj (hunk 4 above)
Comment (SergioFierro) [reply TwilioVerifyDemo/TwilioVerifyDemo.xcodeproj/project.pbxproj]: It's needed, we need to set a provisioning profile with the push capability and with it the team

File: TwilioVerifyDemo/TwilioVerifyDemo.xcodeproj/project.pbxproj (hunk 3 above)
Comment (SergioFierro) [reply TwilioVerifyDemo/TwilioVerifyDemo.xcodeproj/project.pbxproj]: Done

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.38%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.38%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: TwilioVerifyDemo/TwilioVerifyDemo.xcodeproj/project.pbxproj (hunk 2 above)
Comment (yafuquen) [reply TwilioVerifyDemo/TwilioVerifyDemo.xcodeproj/project.pbxproj:596]: Can we set it when generating the app? I am not sure about including a specific provisioning profile (it will be different for developers) or the team id

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.38%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: TwilioVerifyDemo/TwilioVerifyDemo.xcodeproj/project.pbxproj (hunk 2 above)
Comment (SergioFierro) [reply TwilioVerifyDemo/TwilioVerifyDemo.xcodeproj/project.pbxproj]: Done

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.38%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: TwilioVerifyDemo/TwilioVerifyDemo.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -613,7 +613,7 @@
 			buildSettings = {
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
 				CODE_SIGN_ENTITLEMENTS = TwilioVerifyDemo/TwilioVerifyDemo.entitlements;
-				CODE_SIGN_IDENTITY = "iPhone Distribution";
```
Comment (yafuquen) [inline TwilioVerifyDemo/TwilioVerifyDemo.xcodeproj/project.pbxproj:616]: Can we keep distribution? This way developers won't need to use sandbox for push

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.38%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.43%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.38%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.38%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.38%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.38%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.38%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: TwilioVerifyDemo/TwilioVerifyDemo.xcodeproj/project.pbxproj (hunk 6 above)
Comment (SergioFierro) [reply TwilioVerifyDemo/TwilioVerifyDemo.xcodeproj/project.pbxproj]: Done

**Comment by AuthyApps** [comment]
## Code coverage
### Total coverage: **`95.38%`**
File | Coverage
-----|-----
TwilioVerify/TwilioVerify/Sources/Domain/Manager/TwilioVerify.swift | **`82.50%`**
> Powered by [Slather](https://github.com/SlatherOrg/slather)

<p align="right" data-meta="generated_by_UnitTests">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by yafuquen — APPROVED** [review]

**Review by Greshot — APPROVED** [review]
